src/content/blog/2025-01-31-ilp-packet-lifecycle.mdx

```diff
@@ -0,0 +1,88 @@
+All of these packets are sent across the network to the receiver, and must be fulfilled or rejected by the receiver in the second phase in order to be considered as having reached the receiver.
+
+<LargeImg
+  src="/developers/img/blog/2025-01-31/ilp-prepare-spec.png"
+  alt="ILP Prepare Packet Specification"
+/>
+
+### Responding with Fulfill or Reject
+In the second phase of STREAM's execution process, the receiver will need to handle each "prepare" packet with either a fulfillment or a rejection, so that the sender knows whether or not a given "prepare" packet was successfully sent or not.
+The receiver fulfills a "prepare" packet by sending a ["fulfill" packet](https://github.com/interledger/rfcs/blob/main/0027-interledger-protocol-4/0027-interledger-protocol-4.md#ilp-fulfill) back to the sender with the preimage from which the "prepare" packet's condition was generated.
+
+<LargeImg
+  src="/developers/img/blog/2025-01-31/ilp-fulfill-spec.png"
+  alt="ILP Fulfill Packet Specification"
+/>
+
+Since only the receiver can provide the preimage needed to fulfill a "prepare" packet's condition, no node in the network in between it and the sender can intercept the packet and send back a fulfilled packet without it reaching the receiver first.
+As soon as the sender receives enough fulfillments from the receiver to complete the payment, it stops sending packets over STREAM and closes the connection.
+
+If the received "prepare" packet expired before it reached the receiver, the amount received already from prior packets exceeds the total, or some other error occurs, the sender will instead respond with a ["reject" packet](https://github.com/interledger/rfcs/blob/main/0027-interledger-protocol-4/0027-interledger-protocol-4.md#ilp-reject) instead.
+This "reject" packet contains an error code that describes what would have caused the rejection.
+
+<LargeImg
+  src="/developers/img/blog/2025-01-31/ilp-reject-spec.png"
+  alt="ILP Reject Packet Specification"
+/>
+
+On another technical point, there also exist some types of packets that cannot be fulfilled. For instance, an unfulfillable packet gets sent when an Open Payments quote is created whose purpose is to determine what the total cost of the payment will be by factoring any fees or conversion rates that may be applied.
+
+### Wrapping Things Up
+As "fulfill" packets arrive at the sender, it verifies that the preimage sent back in each of them does indeed fulfill the condition sent with the corresponding "prepare" packet.
+Once the STREAM connection finishes sending all of the packets and verifies the corresponding fulfillments from the receiver, it closes. The sending Rafiki instance will then update its outgoing payment as being complete, and then publish a webhook to the receiving Rafiki instance to direct it to mark its corresponding incoming payment as complete as well.
```

Comment:
> and then publish a webhook to the receiving Rafiki instance

the receiving Rafiki is responsible for sending the "incoming payment compelted" webhook to its ASE